expression/testdata/expression_suite_out.json

```diff
@@ -274,6 +274,18 @@
           "    └─TableReader_11(Probe) 10000.00 root  data:TableFullScan_10",
           "      └─TableFullScan_10 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo"
         ]
+      },
+      {
+        "SQL": "explain select * from t1 left join t2 on t1.a = t2.a where ifnull(t2.b, t1.a) = 1",
```

Comment:
```sql
TiDB(root@127.0.0.1:test) > explain select * from t1 left join t2 on t1.a = t2.a where t2.b + t1.a = 1;
+------------------------------+----------+-----------+---------------+--------------------------------------------------------------------------------------------+
| id                           | estRows  | task      | access object | operator info                                                                              |
+------------------------------+----------+-----------+---------------+--------------------------------------------------------------------------------------------+
| HashJoin_7                   | 9990.00  | root      |               | inner join, equal:[eq(test.t1.a, test.t2.a)], other cond:eq(plus(test.t2.b, test.t1.a), 1) |
| ├─TableReader_14(Build)      | 7992.00  | root      |               | data:Selection_13                                                                          |
| │ └─Selection_13             | 7992.00  | cop[tikv] |               | eq(plus(test.t2.b, test.t2.a), 1), not(isnull(test.t2.a))                                  |
| │   └─TableFullScan_12       | 10000.00 | cop[tikv] | table:t2      | keep order:false, stats:pseudo                                                             |
| └─TableReader_11(Probe)      | 9990.00  | root      |               | data:Selection_10                                                                          |
|   └─Selection_10             | 9990.00  | cop[tikv] |               | not(isnull(test.t1.a))                                                                     |
|     └─TableFullScan_9        | 10000.00 | cop[tikv] | table:t1      | keep order:false, stats:pseudo                                                             |
+------------------------------+----------+-----------+---------------+--------------------------------------------------------------------------------------------+
7 rows in set (0.00 sec)
```

this query should also not be converted to `inner join`, please confirm.

Reply:
the `t2.b+t1.a` is null rejective for t2 side. So it can be rewrited to inner join.